weave/trace_server/calls_query_builder.py

```diff
@@ -556,23 +614,36 @@ def _as_sql_base_format(
 
         id_subquery_sql = ""
         if id_subquery_name is not None:
-            id_subquery_sql = f"AND (id IN {id_subquery_name})"
+            id_subquery_sql = f"AND (calls_merged.id IN {id_subquery_name})"
 
         project_param = pb.add_param(self.project_id)
 
         # Special Optimization
         id_mask_sql = ""
         if self.hardcoded_filter and self.hardcoded_filter.filter.call_ids:
-            id_mask_sql = f"AND (id IN {_param_slot(pb.add_param(self.hardcoded_filter.filter.call_ids), 'Array(String)')})"
+            id_mask_sql = f"AND (calls_merged.id IN {_param_slot(pb.add_param(self.hardcoded_filter.filter.call_ids), 'Array(String)')})"
         # TODO: We should also pull out id-masks from the dynamic query
 
+        feedback_join_sql = ""
+        feedback_where_sql = ""
+        if needs_feedback:
+            feedback_where_sql = (
+                f" AND calls_merged.project_id = {_param_slot(project_param, 'String')}"
+            )
+            feedback_join_sql = f"""
+            LEFT JOIN feedback
+            ON (feedback.weave_ref = concat('weave-trace-internal:///', {_param_slot(project_param, 'String')}, '/call/', calls_merged.id))
```

Comment:
any reason to do this concat in the query vs outside and pass it in?

Reply:
i don't think so? but not sure. We have to do a concat either way since the last part is dynamic